src/StreamJsonRpc.Shared/JsonRpc.cs

```diff
@@ -345,8 +427,26 @@ public Task InvokeWithCancellationAsync(string targetName, IReadOnlyList<object>
         /// <exception cref="ObjectDisposedException">If this instance of <see cref="JsonRpc"/> has been disposed.</exception>
         public Task<Result> InvokeWithCancellationAsync<Result>(string targetName, IReadOnlyList<object> arguments = null, CancellationToken cancellationToken = default(CancellationToken))
         {
-            int id = Interlocked.Increment(ref this.nextId);
-            return InvokeCoreAsync<Result>(id, targetName, arguments, cancellationToken);
+            return InvokeWithCancellationCoreAsync<Result>(targetName, arguments, cancellationToken, isParameterObject: false);
```

Comment:
Good point.  I've removed the extra InvokeWithCancellationCoreAsync method and just have the individual methods keep track of incrementing ID and calling the right InvokeCoreAsync method.